### PR TITLE
Support SSD f32/int8 evaluation on COCO dataset

### PR DESCRIPTION
**Pull request by xinyu-intel**
## Description ##
This PR enable ssd training and f32/int8 evaluation on coco2014 dataset.

Changes:

- enable int8 inference on coco2014 dataset on CPU with MKL-DNN backend. Int8 inference achieves 0.253 mAP, which is a comparable accuracy to FP32 (0.2552 mAP).
- replace SoftmaxActivation with softmax since the it has been deprecated.
- support making packed binary file of coco dataset.

@pengzhao-intel @TaoLv @zhreshold 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: example/ssd/README.md [hunk 1]
```diff
@@ -119,7 +120,7 @@ You can use `./demo.py --camera` to use a video capture device with opencv such
 will open a window that will display the camera output together with the detections. You can play
 with the detection threshold to get more or less detections.
 
-### Train the model
+### Train the model on VOC
 * Note that we recommend to use gluon-cv to train the model, please refer to [gluon-cv ssd](https://gluon-cv.mxnet.io/build/examples_detection/train_ssd_voc.html).
 This example only covers training on Pascal VOC dataset. Other datasets should
```
Comment (pengzhao-intel) [inline example/ssd/README.md]: modify this sentence since we already support COCO now :)

File: example/ssd/README.md [hunk 2]
```diff
@@ -166,16 +167,53 @@ Check `python train.py --help` for more training options. For example, if you ha
...
+
+To quantize a model on COCO dataset, follow the [Train instructions](https://github.com/apache/incubator-mxnet/tree/master/example/ssd#train-the-model-on-COCO) to train a FP32 `SSD-VGG16_reduced_300x300` model based on COCO dataset. You can also download our [SSD-VGG16 pre-trained model](http://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/models/ssd_vgg16_reduced_300-7fedd4ad.zip) and [packed binary data](http://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/ssd_coco-val-e91096e8.zip). Create `model` and `data` directories if they're not exist, extract the zip files, then rename the uncompressed files as follows (eg, rename `ssd_coco-val-e91096e8.idx` to `val.idx`, `ssd_coco-val-e91096e8.lst` to `val.lst`, `ssd_coco-val-e91096e8.rec` to `val.rec`, `ssd_vgg16_reduced_300-7fedd4ad.params` to `ssd_vgg16_reduced_300-0000.params`, `ssd_vgg16_reduced_300-symbol-7fedd4ad.json` to `ssd_vgg16_reduced_300-symbol.json`.)
```
Comment (pengzhao-intel) [inline example/ssd/README.md]: what's the difference between two paragrams of VOC and COCO? 
Could you merge into one?

**Comment by zhreshold** [comment]
lgtm, will wait until ci passes

File: example/ssd/README.md (hunk 2 above)
Comment (xinyu-intel) [reply example/ssd/README.md]: different link

File: example/ssd/README.md (hunk 1 above)
Comment (xinyu-intel) [reply example/ssd/README.md]: ok

**Comment by pengzhao-intel** [comment]
Thank you for your contribution. Merging now.
